Sources/Auth/Facebook/FacebookAuthenticator.swift

```diff
@@ -12,58 +12,65 @@ import PovioKit
 import PovioKitAuthCore
 import PovioKitPromise
 
-public protocol FacebookAuthProvidable {
-  typealias Authorized = Bool
-  typealias Response = AuthProvider.Response
-  
-  func signIn(from presentingViewController: UIViewController) -> Promise<Response>
-  static func signOut()
-  static func isAuthorized() -> Authorized
+public protocol FacebookAuthProvidable: AuthProvidable {
+  func signIn(from presentingViewController: UIViewController,
+              with extraPermissions: [Permission]) -> Promise<Response>
+  var isAuthenticated: Authenticated { get }
```

Comment:
Does this and other such protocols add any value? For instance, shouldn't `var isAuthenticated: Authenticated { get }` be part of all `Authenticator`s?

Reply:
The thing is, all but AppleAuthenticator has a state for it. Only Apple requires to query it. Therefore, we cannot use the same getter for all instances.

Reply:
How about converting it into a `Promise`? Those that don't need to query can return `.value(state)`, whereas for Apple's you do the querying ..

Reply:
Not perfect. How would a dev know where to pick which?

Reply:
What I mean is that there is only one getter.

```swift
protocol AuthProvidable {
  ...
  var authenticated: Promise<Bool> { get }
}
```

Reply:
I understand. But not optimal. It's not optimal in any way either. I had this on initial implementation but abandoned it later. I'll revisit.

Reply:
Just a suggestion. :)